data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -62,23 +64,30 @@ public AadAuthenticationHelper(@NotNull ConnectionStringBuilder csb) throws URIS
     }
 
     String acquireAccessToken() throws DataServiceException  {
+        if(lastAuthenticationResult != null && lastAuthenticationResult.getExpiresOnDate().after(new Date())){
```

Comment:
What will happen if we have a really close expiration (but still valid) and by the time we issue the call to the service the token will expire?
Isn't it safer to have a buffer of a few seconds or a minute from the actual expiration to trigger a refresh of the cached one?

Reply:
will do 1 minute